app/uk/gov/hmrc/incometaxpenaltiesfrontend/controllers/PenaltyCalculationController.scala

```diff
@@ -30,11 +30,13 @@ class PenaltyCalculationController @Inject()(override val controllerComponents:
                                              penaltyCalculationView: PenaltyCalculation,
                                              authActions: AuthActions)(implicit appConfig: AppConfig) extends FrontendBaseController with I18nSupport {
 
-  val penaltyCalculationPage: Action[AnyContent] =
-    authActions.asMTDUserOld() { implicit currentUserRequest =>
+  def penaltyCalculationPage(isAgent:Boolean): Action[AnyContent] = authActions.asMTDUserOld() { implicit currentUserRequest =>
+    if(currentUserRequest.isAgent != isAgent) {
+      Redirect(routes.IndexController.homePage(currentUserRequest.isAgent))
+    }else{
       Ok(penaltyCalculationView(currentUserRequest.isAgent))
+    }
```

Comment:
Same as above If you call authActions.asMTDUserWithPenaltyData(isAgent) rather than the authActions.asMTDUserOldWithPenaltyData() then you won't need to do the redirect to home url stuff as the actions in there handle if an agent hits an individual endpoint and vice versa.